src/plugins/discover/public/application/main/components/field_stats_table/field_stats_table.tsx

```diff
@@ -50,163 +46,61 @@ export const FieldStatisticsTable = (props: FieldStatisticsTableProps) => {
-    []
-  );
-  // Embeddable state exposed to external consumers like Discover
-  const embeddableApiSubscription = useRef<Subscription | undefined>(undefined);
-
   const showPreviewByDefault = useMemo(
     () => (stateContainer ? !stateContainer.appState.getState().hideAggregatedPreview : true),
     [stateContainer]
   );
 
-  useEffect(
-    function syncChangesToEmbeddableState() {
-      const availableFields$ = stateContainer?.dataState.data$.availableFields$;
-
-      const refetch = stateContainer?.dataState.refetch$.subscribe(() => {
-        if (embeddableState$) {
-          embeddableState$.next({
-            ...embeddableState$.getValue(),
-            lastReloadRequestTime: Date.now(),
-          });
-        }
-      });
-
-      const fields = availableFields$?.subscribe(() => {
-        if (embeddableState$ && !availableFields$?.getValue().error) {
-          embeddableState$.next({
-            ...embeddableState$.getValue(),
-            fieldsToFetch: availableFields$?.getValue().fields,
-          });
-        }
-      });
-
-      return () => {
-        refetch?.unsubscribe();
-        fields?.unsubscribe();
-        embeddableState$?.unsubscribe();
-        embeddableApiSubscription.current?.unsubscribe();
-      };
-    },
-    [embeddableState$, stateContainer]
-  );
-
-  const initialState = useMemo(() => {
-    return {
-      shouldGetSubfields: true,
-      dataView,
-      savedSearch,
-      query,
-      visibleFieldNames: columns,
-      sessionId: searchSessionId,
-      fieldsToFetch: stateContainer?.dataState.data$.availableFields$?.getValue().fields,
-      totalDocuments,
-      showPreviewByDefault,
-      samplingOption: {
-        mode: 'normal_sampling',
-        shardSize: 5000,
-        seed: searchSessionId,
-      } as NormalSamplingOption,
-    };
-    // We just need a snapshot of initialState upon table mounting
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
-  useEffect(() => {
-    if (embeddableState$) {
-      // Update embeddable whenever one of the important input changes
-      embeddableState$.next({
-        shouldGetSubfields: true,
-        dataView,
-        savedSearch,
-        query,
-        visibleFieldNames: columns,
-        sessionId: searchSessionId,
-        fieldsToFetch: stateContainer?.dataState.data$.availableFields$?.getValue().fields,
-        totalDocuments,
-        showPreviewByDefault,
-        samplingOption: {
-          mode: 'normal_sampling',
-          shardSize: 5000,
-          seed: searchSessionId,
-        } as NormalSamplingOption,
-      });
-    }
-  }, [
-    embeddableState$,
-    dataView,
-    savedSearch,
-    query,
-    columns,
-    filters,
-    searchSessionId,
-    totalDocuments,
-    stateContainer,
-    showPreviewByDefault,
-  ]);
+  const lastReloadRequestTime$ = stateContainer?.dataState?.refetch$
+    ? stateContainer?.dataState?.refetch$.pipe(map(() => Date.now()))
```

Comment:
Nit: since this creates a new observable on each render, I think it will resubscribe in `useObservable` every time.

Reply:
Thanks for catching that. Updated here https://github.com/elastic/kibana/pull/181502/commits/e6603b563419e44e0d45bff8d663d14d86fa91cf.